src/tools.cpp

```diff
@@ -725,6 +725,16 @@ std::string getCombatName(CombatType_t combatType)
 	return "unknown";
 }
 
+CombatType_t getCombatTypeByName(std::string name)
+{
+	for (auto it = combatTypeNames.begin(); it != combatTypeNames.end(); ++it) {
```

Comment:
Use `find_if`, makes it clearer you just want to find an element in the map:
```
auto it = std::find_if(combatTypeNames.begin(), combatTypeNames.end(),
  [&](std::pair<std::string, CombatType_t> const& pair)
  {
    return pair.second == name;
  });

if (it == combatTypeNames.end())
  return COMBAT_NONE;
else
  return it->first;
```

Alternatively, there should be a reverse-lookup map.

Reply:
Great modification, ty. Just a correction:
`[&](std::pair<std::string, CombatType_t> const& pair)` 
modify to:
`[&](std::pair<CombatType_t, std::string> const& pair)`